process_qsos.m

```diff
@@ -405,9 +400,10 @@
         dla_M      = this_M      .* absorption;
         dla_omega2 = this_omega2 .* absorption.^2;
         
+        % Add a penalty for short spectra: the expected reduced chi^2 of each spectral pixel that would have been observed.
         this_sample_log_likelihoods_dla(1, i) = ...
             log_mvnpdf_low_rank(this_flux, dla_mu, dla_M, ...
-            dla_omega2 + this_noise_variance);
+            dla_omega2 + this_noise_variance) - lambda_not_observed/ dlambda;
```

Comment:
- Does this implementation corresponding to some manipulations in the likelihoods? Like dividing the number of pixels in the likelihood? In that case, we probably need to subtract in log like `- log(lambda_not_observed / dlambda)`. 

I was thinking this for a while, maybe you want this:
```
P(D | θ) := P(D | θ) *  num_pixels_observed / (num_pixels) 
= P(D | θ) *  num_pixels_observed / (num_pixels_observed + num_pixels_not_observed) 
```

results in:

```matlab
lambda_observed = (max_observed_lambda - min_observed_lambda);

this_sample_log_likelihoods_dla(1, i) = ...
            log_mvnpdf_low_rank(this_flux, dla_mu, dla_M, ...
            dla_omega2 + this_noise_variance) + log(lambda_observed / dlambda) ...
           - log(lambda_observed / dlambda + lambda_not_observed / dlambda)
```

- Do you think we need to apply this penalty at `sample_log_likelihoods_no_dla`?

Reply:
Yes! Good points, thank you! The second term
 - log(lambda_observed / dlambda + lambda_not_observed / dlambda)
is constant, is it not? So we can drop it.

Reply:
> log(lambda_observed / dlambda + lambda_not_observed / dlambda) is constant, is it not? So we can drop it.

Ah yes! In this case, instead of penalising missing pixels, we encourage the spectra with more pixels to have a higher likelihood. 

One issue I could imagine is if `the_sample_log_likelihoods_dla` becomes larger than zero, then `model_posteriors` would be messed up. But that probably not happen too much in the z code since lots of models are very poorly fitted I think.